src/Core/src/Platform/Windows/SearchBarExtensions.cs

```diff
@@ -132,5 +132,35 @@ public static void UpdateIsTextPredictionEnabled(this AutoSuggestBox platformCon
 		{
 			// AutoSuggestBox does not support this property
 		}
+
+		private static readonly string[] CancelButtonColorKeys =
+		{
+			"TextControlButtonForeground",
+			"TextControlButtonForegroundPointerOver",
+			"TextControlButtonForegroundPressed",
+		};
+
+		internal static void UpdateCancelButtonColor(this AutoSuggestBox platformControl, ISearchBar searchBar)
+		{
+			var cancelButton = platformControl.GetDescendantByName<Button>("DeleteButton");
+
+			if (cancelButton == null)
+				return;
+
+			var cancelButtonColor = searchBar.CancelButtonColor?.ToPlatform();
+
+			if (cancelButtonColor == null)
+			{
+				cancelButton.Resources.RemoveKeys(CancelButtonColorKeys);
+				cancelButton.Foreground = null;
```

Comment:
Can we create a version of `ButtonExtensions.UpdateTextColor` that takes the resourcekeys as a parameter?  It seems like we're duplicating the logic here that we already have there. 

I also noticed that in `ButtonExtensions.UpdateTextColor`  we clear the property whereas here it's setting foreground to null. 

I'm guessing the `ButtonExtensions` one is more correct?

Reply:
Is a good idea. I have created a new extension method in `ButtonExtensions `and reuse it with the Button and in this case with the SearchBar clear button.